ts/packages/anchor/src/provider.ts

```diff
@@ -129,35 +153,46 @@ export class AnchorProvider implements Provider {
    * @param opts    Transaction confirmation options.
    */
   async sendAndConfirm(
-    tx: Transaction | VersionedTransaction,
-    signers?: Signer[],
-    opts?: ConfirmOptions
+    tx: Transaction,
+    signers?: CryptoKeyPair[],
+    opts?: SendAndConfirmOptions
   ): Promise<TransactionSignature> {
     if (opts === undefined) {
       opts = this.opts;
     }
 
-    if (isVersionedTransaction(tx)) {
-      if (signers) {
-        tx.sign(signers);
-      }
-    } else {
-      tx.feePayer = tx.feePayer ?? this.wallet.publicKey;
-      tx.recentBlockhash = (
-        await this.connection.getLatestBlockhash(opts.preflightCommitment)
-      ).blockhash;
-
-      if (signers) {
-        for (const signer of signers) {
-          tx.partialSign(signer);
-        }
+    // Versioned transactions now have first-class support,
+    // so we no longer need to differentiate.
+    const blockhashLifetime = (
+      await this.rpc
+        .getLatestBlockhash({ ...opts } as GetBlockhashOptions)
+        .send()
+    ).value;
+
+    // We don't have a helper for this yet.
+    // See solana-labs/solana-web3.js#1573
+    const txWithFeePayer = setTransactionFeePayer(this.wallet.address, tx);
+    const txWithFeePayerAndBlockhashLifetime =
+      setTransactionLifetimeUsingBlockhash(blockhashLifetime, txWithFeePayer);
+    let resultTx: Parameters<typeof signTransaction>[1] = {
+      ...txWithFeePayerAndBlockhashLifetime,
+    };
+    let currentTx: Parameters<typeof signTransaction>[1] = {
+      ...txWithFeePayerAndBlockhashLifetime,
+    };
```

Comment:
☠️

It could make sense to split up the massive `Connection` object for the people who care about bundle size but adding a separate function for each transaction method seems a bit odd to me given most of these functions are going to be used pretty often and would only be couple hundred bytes at most.

I think this will significantly slow down developers trying to find which function to import next.

There is also this aspect of reducing the bundle size but decreasing the performance with excessive cloning each time(from my limited view of the PR).

Reply:
Agreed the cloning is not ideal but once the helpers are shipped this will look nothing like the implementation I’ve written here. 

It will instead look something like this:

```typescript
const finalTx = pipe(
  tx,
  tx => setTransactionFeePayer(feePayer, tx),
  tx => setTransactionLifetimeUsingBlockhash(lifetime, tx)
):
```

Reply:
Then you might have one other async call like:

```typescript
const signedTx = await signTransaction(signers, tx);
```

Reply:
> I think this will significantly slow down developers trying to find which function to import next.

Thank you for this feedback; you're not wrong! Fluent interfaces are brutal for bundle size, but good for discoverability, as you've pointed out. We'll have to give a good think to how to reduce this pain.